packages/openapi-v3/src/consolidate-schema.ts

```diff
@@ -0,0 +1,106 @@
+import compare from 'json-schema-compare';
+import _ from 'lodash';
+import {
+  ISpecificationExtension,
+  isSchemaObject,
+  OpenApiSpec,
+  PathsObject,
+  ReferenceObject,
+  SchemaObject,
+} from './types';
+
+let $schemaRefs: {
+  [schema: string]: SchemaObject | ReferenceObject;
+};
+let $paths: PathsObject;
+
+/*
+ *  Recursively search OpenApiSpec PathsObject for SchemaObjects with title property.
+ *  Move reusable schema bodies to #/components/schemas and replace with json pointer.
+ *  Handles collisions /w title, schema body pair comparision.
+ */
+export function ConsolidateSchemaObjects(spec: OpenApiSpec): OpenApiSpec {
```

Comment:
We use camelCase names for functions.

```suggestion
export function consolidateSchemaObjects(spec: OpenApiSpec): OpenApiSpec {
```